app/src/views/features/rules/RuleEditor/components/RuleDetailsPanel/RuleDetailsPanel.tsx

```diff
@@ -57,65 +61,87 @@ export const RuleDetailsPanel: React.FC<RuleDetailsPanelProps> = ({ ruleType, so
+            {RULE_DETAILS[ruleType].useCases?.length > 0 &&
+              RULE_DETAILS[ruleType].useCases?.map(({ useCase, example }, index) => {
+                return (
+                  <>
                     <li key={index} className="use-case-list-item">
+                      <MdOutlineAssignmentTurnedIn className="list-item-icon" />
                       <div className="use-case">
-                        {useCase}
+                        {useCase} {"  "}
                         <br />
-                        {suggestedTemplateId ? (
-                          <Button
-                            type="link"
-                            className="link use-template-btn"
-                            icon={<MdOutlineFactCheck className="anticon" />}
-                            onClick={() => handleUseTemplateClick(suggestedTemplateId)}
-                          >
-                            Use template
-                          </Button>
+                        {example?.type === ExampleType.USE_TEMPLATE ? (
+                          <>
+                            <Button
+                              type="link"
+                              className="link use-template-btn"
+                              icon={<MdOutlineFactCheck className="anticon" />}
+                              onClick={() => handleUseTemplateClick(example.suggestedTemplateId)}
+                            >
+                              Use template
+                            </Button>
+                          </>
+                        ) : example?.type === ExampleType.DOWNLOAD_DESKTOP_APP ? (
+                          <>
+                            <Button
+                              type="link"
+                              target="_blank"
+                              href={example.link}
+                              className="link use-template-btn"
+                              icon={<MdOutlineFileDownload className="anticon" />}
+                              onClick={() => {
+                                // TODO: send event
+                              }}
+                            >
+                              Download Desktop App
+                            </Button>
+                          </>
+                        ) : example?.type === ExampleType.PLAYGROUND_LINK ? (
+                          <>
+                            <Button
+                              type="link"
+                              target="_blank"
+                              href={example.link}
+                              className="link use-template-btn"
+                              icon={<MdOutlineOpenInNew className="anticon" />}
+                              onClick={() => {
+                                // send event
+                              }}
+                            >
+                              Try this on Requestly Playground
+                            </Button>
+                          </>
```

Comment:
Can we avoid this chaining? makes it difficult to read. Can we use switch case to conditionally render the buttons.

Reply:
Switch case wont work inside return